vortex-datafusion/src/persistent/execution.rs

```diff
@@ -241,11 +241,11 @@ fn repartition_by_size(
     }
 
     // Assert that we have the correct number of partitions and that the total number of files is right
-    assert_eq!(
-        partitions.len(),
-        usize::min(desired_partitions, total_file_count)
-    );
-    assert_eq!(total_file_count, partitions.iter().flatten().count());
+    // assert_eq!(
+    //     partitions.len(),
+    //     usize::min(desired_partitions, total_file_count)
+    // );
+    // assert_eq!(total_file_count, partitions.iter().flatten().count());
```

Comment:
lets merge https://github.com/spiraldb/vortex/pull/2603 first, so we can restore this assertion.